diff --git a/node_modules/react-native-yamap/android/build.gradle b/node_modules/react-native-yamap/android/build.gradle
index d9de295..a02f39c 100644
--- a/node_modules/react-native-yamap/android/build.gradle
+++ b/node_modules/react-native-yamap/android/build.gradle
@@ -10,6 +10,7 @@ apply plugin: 'com.android.library'
 apply plugin: 'org.jetbrains.kotlin.android'
 
 android {
+    namespace 'ru.vvdev.yamap'
 
     defaultConfig {
         compileSdkVersion 34
diff --git a/node_modules/react-native-yamap/android/src/main/AndroidManifest.xml b/node_modules/react-native-yamap/android/src/main/AndroidManifest.xml
index 330157d..5f5d7a2 100644
--- a/node_modules/react-native-yamap/android/src/main/AndroidManifest.xml
+++ b/node_modules/react-native-yamap/android/src/main/AndroidManifest.xml
@@ -1,5 +1,4 @@
 
-<manifest xmlns:android="http://schemas.android.com/apk/res/android"
-          package="ru.vvdev.yamap">
+<manifest xmlns:android="http://schemas.android.com/apk/res/android">
 
 </manifest>
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt
index 54d8a0f..d4a9d34 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/ClusteredYamapViewManager.kt
@@ -19,7 +19,7 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
         return REACT_CLASS
     }
 
-    override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
+    override fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .build()
     }
@@ -128,7 +128,7 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
             }
 
             "findRoutes" -> if (args != null) {
-                findRoutes(view, args.getArray(0), args.getArray(1), args.getString(2))
+                findRoutes(view, args.getArray(0) ?: return, args.getArray(1) ?: return, args.getString(2) ?: "")
             }
 
             "setZoom" -> if (args != null) {
@@ -152,11 +152,11 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
             }
 
             "getScreenPoints" -> if (args != null) {
-                view.emitWorldToScreenPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitWorldToScreenPoints(it, args.getString(1) ?: "") }
             }
 
             "getWorldPoints" -> if (args != null) {
-                view.emitScreenToWorldPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitScreenToWorldPoints(it, args.getString(1) ?: "") }
             }
 
             else -> throw IllegalArgumentException(
@@ -171,7 +171,8 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
 
     @ReactProp(name = "clusteredMarkers")
     fun setClusteredMarkers(view: View, points: ReadableArray) {
-        castToYaMapView(view).setClusteredMarkers(points.toArrayList())
+        @Suppress("UNCHECKED_CAST")
+        castToYaMapView(view).setClusteredMarkers(points.toArrayList().filterNotNull() as ArrayList<Any>)
     }
 
     @ReactProp(name = "clusterColor")
@@ -243,7 +244,7 @@ class ClusteredYamapViewManager internal constructor() : ViewGroupManager<Cluste
             val vehicles = ArrayList<String>()
             if (jsVehicles != null) {
                 for (i in 0 until jsVehicles.size()) {
-                    vehicles.add(jsVehicles.getString(i))
+                    jsVehicles.getString(i)?.let { vehicles.add(it) }
                 }
             }
             castToYaMapView(view).findRoutes(points, vehicles, id)
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapCircleManager.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapCircleManager.kt
index 85be269..b96b06d 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapCircleManager.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapCircleManager.kt
@@ -14,7 +14,7 @@ class YamapCircleManager internal constructor() : ViewGroupManager<YamapCircle>(
         return REACT_CLASS
     }
 
-    override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
+    override fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .put("onPress", MapBuilder.of("registrationName", "onPress"))
             .build()
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt
index e32d22c..c6a768b 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapMarkerManager.kt
@@ -17,7 +17,7 @@ class YamapMarkerManager internal constructor() : ViewGroupManager<YamapMarker>(
         return REACT_CLASS
     }
 
-    override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
+    override fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .put("onPress", MapBuilder.of("registrationName", "onPress"))
             .build()
@@ -109,8 +109,8 @@ class YamapMarkerManager internal constructor() : ViewGroupManager<YamapMarker>(
             "animatedMoveTo" -> {
                 val markerPoint = args!!.getMap(0)
                 val moveDuration = args.getInt(1)
-                val lon = markerPoint.getDouble("lon").toFloat()
-                val lat = markerPoint.getDouble("lat").toFloat()
+                val lon = markerPoint?.getDouble("lon")?.toFloat() ?: 0f
+                val lat = markerPoint?.getDouble("lat")?.toFloat() ?: 0f
                 val point = Point(lat.toDouble(), lon.toDouble())
                 castToMarkerView(view).animatedMoveTo(point, moveDuration.toFloat())
                 return
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolygonManager.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolygonManager.kt
index cf14b19..630ddbd 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolygonManager.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolygonManager.kt
@@ -15,7 +15,7 @@ class YamapPolygonManager internal constructor() : ViewGroupManager<YamapPolygon
         return REACT_CLASS
     }
 
-    override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
+    override fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .put("onPress", MapBuilder.of("registrationName", "onPress"))
             .build()
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolylineManager.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolylineManager.kt
index 8be99c8..2cbbe6e 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolylineManager.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapPolylineManager.kt
@@ -15,7 +15,7 @@ class YamapPolylineManager internal constructor() : ViewGroupManager<YamapPolyli
         return REACT_CLASS
     }
 
-    override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
+    override fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .put("onPress", MapBuilder.of("registrationName", "onPress"))
             .build()
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt
index 8b05905..438c6d2 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/YamapViewManager.kt
@@ -19,7 +19,7 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
         return REACT_CLASS
     }
 
-    override fun getExportedCustomDirectEventTypeConstants(): Map<String, Any>? {
+    override fun getExportedCustomDirectEventTypeConstants(): MutableMap<String, Any>? {
         return MapBuilder.builder<String, Any>()
             .build()
     }
@@ -128,7 +128,7 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
             }
 
             "findRoutes" -> if (args != null) {
-                findRoutes(view, args.getArray(0), args.getArray(1), args.getString(2))
+                findRoutes(view, args.getArray(0) ?: return, args.getArray(1) ?: return, args.getString(2) ?: "")
             }
 
             "setZoom" -> if (args != null) {
@@ -152,11 +152,11 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
             }
 
             "getScreenPoints" -> if (args != null) {
-                view.emitWorldToScreenPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitWorldToScreenPoints(it, args.getString(1) ?: "") }
             }
 
             "getWorldPoints" -> if (args != null) {
-                view.emitScreenToWorldPoints(args.getArray(0), args.getString(1))
+                args.getArray(0)?.let { view.emitScreenToWorldPoints(it, args.getString(1) ?: "") }
             }
 
             else -> throw IllegalArgumentException(
@@ -237,7 +237,7 @@ class YamapViewManager internal constructor() : ViewGroupManager<YamapView>() {
 
             if (jsVehicles != null) {
                 for (i in 0 until jsVehicles.size()) {
-                    vehicles.add(jsVehicles.getString(i))
+                    jsVehicles.getString(i)?.let { vehicles.add(it) }
                 }
             }
 
diff --git a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt
index 48c1cdf..7982851 100644
--- a/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt
+++ b/node_modules/react-native-yamap/android/src/main/java/ru/vvdev/yamap/view/YamapView.kt
@@ -226,7 +226,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
 
         for (i in 0 until worldPoints.size()) {
             val p = worldPoints.getMap(i)
-            val worldPoint = Point(p.getDouble("lat"), p.getDouble("lon"))
+            val worldPoint = Point(p?.getDouble("lat") ?: 0.0, p?.getDouble("lon") ?: 0.0)
             val screenPoint = mapWindow.worldToScreen(worldPoint)
             screenPoints.pushMap(screenPointToJSON(screenPoint))
         }
@@ -245,7 +245,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
 
         for (i in 0 until screenPoints.size()) {
             val p = screenPoints.getMap(i)
-            val screenPoint = ScreenPoint(p.getDouble("x").toFloat(), p.getDouble("y").toFloat())
+            val screenPoint = ScreenPoint((p?.getDouble("x") ?: 0.0).toFloat(), (p?.getDouble("y") ?: 0.0).toFloat())
             val worldPoint = mapWindow.screenToWorld(screenPoint)
             worldPoints.pushMap(worldPointToJSON(worldPoint))
         }
@@ -626,7 +626,7 @@ open class YamapView(context: Context?) : MapView(context), UserLocationObjectLi
         val routeMetadata = Arguments.createMap()
         val routeWeightData = Arguments.createMap()
         val sectionWeightData = Arguments.createMap()
-        val transports: MutableMap<String, ArrayList<String?>?> = HashMap()
+        val transports: MutableMap<String, ArrayList<String?>> = HashMap()
         routeWeightData.putString("time", routeWeight.time.text)
         routeWeightData.putInt("transferCount", routeWeight.transfersCount)
         routeWeightData.putDouble("walkingDistance", routeWeight.walkingDistance.value)
